miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java

```diff
@@ -820,7 +828,13 @@ public String processSubmit(@ModelAttribute("sample") Sample sample, ModelMap mo
       if (builder.getTissueClass() != null) {
         builder.setTissueClass(sampleClassService.get(builder.getTissueClass().getId()));
       }
-      sample = builder.build();
+      sample = builder.build(new InferStockFromAliquot() {
+
+        @Override
+        public SampleClass infer(SampleClass sampleClass) {
+          return sampleClassService.inferStockFromAliquot(sampleClass);
```

Comment:
Fixed. The performance implications of creating the extra object are negligible.